langchain/retrievers/svm.py

```diff
@@ -52,9 +53,23 @@ def get_relevant_documents(self, query: str) -> List[Document]:
         similarities = clf.decision_function(x)
         sorted_ix = np.argsort(-similarities)
 
+        # svm.LinearSVC in scikit-learn is non-deterministic.
+        # if a text is the same as a query, there is no guarantee the query will be in the first index.
+        # this performs a simple swap, this works because anything left of the 0 should be equivalent.
+        zero_index = np.where(sorted_ix == 0)[0][0]
```

Comment:
Would be helpful to annotate the axes (what it means to take [0][0])

Reply:
`np.where(arr == 0)` returns a tuple of arrays containing the indices where the condition `arr == 0` is satisfied. This is a tuple with one element, which is an array containing the index where the condition is true (where the value is 0). 

To get the actual index value where the condition is true, we use indexing with `[0][0]`. The first `[0]` selects the first element of the tuple (which is the array with the index where the condition is true), and the second `[0]` selects the first (and only) element of that array. So `np.where(arr == 0)[0][0]` will return the index of the first occurrence of `0` in the array.